src/Compiler/Driver/GraphChecking/FileContentMapping.fs

```diff
@@ -619,32 +619,36 @@ let visitSynMemberSig (ms: SynMemberSig) : FileContentEntry list =
     | SynMemberSig.NestedType _ -> []
 
 let mkFileContent (f: FileInProject) : FileContentEntry list =
-    [
-        match f.ParsedInput with
-        | ParsedInput.SigFile (ParsedSigFileInput (contents = contents)) ->
-            for SynModuleOrNamespaceSig (longId = longId; kind = kind; decls = decls; attribs = attribs) in contents do
-                yield! List.collect visitSynAttributeList attribs
-
-                match kind with
-                | SynModuleOrNamespaceKind.GlobalNamespace
-                | SynModuleOrNamespaceKind.AnonModule -> yield! List.collect visitSynModuleSigDecl decls
-                | SynModuleOrNamespaceKind.DeclaredNamespace ->
-                    let path = longIdentToPath false longId
-                    yield FileContentEntry.TopLevelNamespace(path, List.collect visitSynModuleSigDecl decls)
-                | SynModuleOrNamespaceKind.NamedModule ->
-                    let path = longIdentToPath true longId
-                    yield FileContentEntry.TopLevelNamespace(path, List.collect visitSynModuleSigDecl decls)
-        | ParsedInput.ImplFile (ParsedImplFileInput (contents = contents)) ->
-            for SynModuleOrNamespace (longId = longId; attribs = attribs; kind = kind; decls = decls) in contents do
-                yield! List.collect visitSynAttributeList attribs
-
-                match kind with
-                | SynModuleOrNamespaceKind.GlobalNamespace
-                | SynModuleOrNamespaceKind.AnonModule -> yield! List.collect visitSynModuleDecl decls
-                | SynModuleOrNamespaceKind.DeclaredNamespace ->
-                    let path = longIdentToPath false longId
-                    yield FileContentEntry.TopLevelNamespace(path, List.collect visitSynModuleDecl decls)
-                | SynModuleOrNamespaceKind.NamedModule ->
-                    let path = longIdentToPath true longId
-                    yield FileContentEntry.TopLevelNamespace(path, List.collect visitSynModuleDecl decls)
-    ]
+    if f.Idx = 0 then
```

Comment:
I would prefer to let the caller handle the first file (and not call this function at all for it), as it better placed to make index-specific decisions as it handles all files at the same time.